ompi/mca/pml/base/pml_base_select.c

```diff
@@ -359,22 +348,26 @@ mca_pml_base_pml_check_selected(const char *my_pml,
      */
     if (NULL == remote_pml) {
         opal_output_verbose( 10, ompi_pml_base_framework.framework_output,
-                            "check:select: got a NULL pml from rank=0");
+                            "check:select: got a NULL pml from process %s",
+                            OMPI_NAME_PRINT(&proc_name));
         return OMPI_ERR_UNREACH;
     }
 
     opal_output_verbose( 10, ompi_pml_base_framework.framework_output,
-                        "check:select: checking my pml %s against rank=0 pml %s",
-                        my_pml, remote_pml);
+                        "check:select: checking my pml %s against process %s"
+                        " pml %s", my_pml, OMPI_NAME_PRINT(&proc_name),
+                        remote_pml);
 
     /* if that module doesn't match my own, return an error */
     if ((size != strlen(my_pml) + 1) ||
         (0 != strcmp(my_pml, remote_pml))) {
-        char *errhost = opal_get_proc_hostname(&procs[0]->super);
+        char *errhost;
+        OPAL_MODEX_RECV_VALUE(ret, PMIX_HOSTNAME, &proc_name,
```

Comment:
This needs to be an "optional" recv as the hostname is not always available.

Reply:
Yes, ill make it optional. Thank you